### DISCUSSION TITLE
v39.0-beta1

### DISCUSSION THREAD

**Discussion by westnordost** [streetcomplete/StreetComplete#3626] [Release Discussions]
Finally, a new release! You know how it is, the holidays... But this update should you give plenty to do, for now you can record the surface quality of roads!

  <h3>New Quests</h3>
  <ul>
  <li><i>Does this road have a shoulder?</i> (#2444, #3613)</li>
  <li><i>What is the surface quality of this road here?</i> (#3617, #3257, #1630), special thanks to @Helium314 but also to @mcliquid, @NicoHood, @rhhsm, @mnalis, @1ec5 and many more</li>
  <li><i>Where is this fire hydrant located?</i> (#3368), by @thefeiter</li>
  </ul>
  <h3>Other enhancements</h3>
  <ul>
  <li>You can now answer that a building has a house name <i>and</i> a number (#1983, #3582), by @matkoniecz</li>
  <li>Some improvements on the fire hydrant quest(s) (#3368), by @thefeiter, (#3601)</li>
  <li>Some minor enhancements and clearer wordings (#3576, #3610, ...), thanks @matkoniecz</li>
  <li>Added "Lifesaver" achievement (#3616), by @FloEdelmann</li>
  </ul>
  <h3>Fixes</h3>
  <ul>
  <li>Fix rarely quest answers don't get uploaded automatically until next restart of app (#3494)</li>
  </ul>

<hr /><em>This discussion was created from the release <a href='https://github.com/streetcomplete/StreetComplete/releases/tag/'>v39.0-beta1</a>.</em>

**Comment by mnalis** [comment]
FYI some Emojis are missing for smoothness quest on my older phone: Xiaomi Redmi Note 3 pro (OS: MIUI 10.2.1.0  - Android 6.0.1)
![small_Screenshot_2022-01-09-12-57-20-670_de westnordost streetcomplete](https://user-images.githubusercontent.com/156656/148681655-d8f34426-6848-43bf-8665-763e0d5ca625.png)

Not really a newest phone and I use it mostly only for testing and occasional navigation /  mapping, but amazingly Xiaomi still published some security updates for it a month ago. (Unfortunately I can't try beta on my newer phone as it uses f-droid signed version, and I'd like to avoid losing all the presets I've setup there)

**Reply by westnordost** [reply]
Yes, it's known. Though, I don't find it so important as the issue will solve itself eventually anyway.  The 🛴 displays correctly on Android 7+ (2016), the 🛹 displays correctly on Android 9+ (2018).

**Reply by mnalis** [reply]
Not a big deal deal for me personally of course, but it might confuse newer users. Do we have any idea how many users use lower than Android 7/9 (I don't know if google play makes that information available to publishers?)

**Reply by westnordost** [reply]
It's available, for users who installed the app via Google Play. This is the % of people who have **installed** it, the percentages will be different who actually opens it (regularly)

| Android version | % of users |
| --------------- | ---------- |
| 5.1 | 0.2% |
| 6.0 | 0.7% |
| 7.1 | 0.8% |
| 8.0 | 3.1% |
| 8.1 | 1.9% |

So, 0.9% can not display the 🛴 , 6.7% can not display the 🛹

**Reply by FloEdelmann** [reply]
Pickup truck requires Android 11: https://emojipedia.org/pickup-truck/

**Reply by rhhsm** [reply]
I see more reasons than already mentioned to replace the icons by the text "usable by VEHICLE"...

**Reply by mnalis** [reply]
I've compiled it in my fork to be able to test it on my relatively new main phone (Huawei P30 Pro, EMUI 11.0, Android 10) and the emoji also broke, albeit at a different place:
![small_Screenshot_20220110_192130_de westnbordost streetcomplete mn debug](https://user-images.githubusercontent.com/156656/148828035-10f3b19b-a390-4508-8ea7-204e273f94df.jpg)

I'll agree with @rhhsm here: I've voted then for emoji-only variant assuming they would display correctly at 99% of the phones as it saves screen estate; but if emojis are not really that reliable (and it seems like they aren't) then some alternative like a text only version (or emojis converted to images, or embedded font or what have you) might perhaps be a better way to go. Especially for SC which is placed high on "nice looks" scale.

**Reply by FloEdelmann** [reply]
I started an implementation in #3630.

**Reply by westnordost** [reply]
That 🛻 doesn't render doesn't matter, because it already says right there "usable only by off-road vehicles".

**Reply by mnalis** [reply]
Yes, it says what the picture shows (which is why I agreed with one alternative of simply removing all emojis and replacing them with text). But broken emojis still make (otherwise graphically quite pleasing so far!) app **look ugly and broken**, which _should_ matter IMHO.

**Comment by thefeiter** [comment]
>  Some improvements on the fire hydrant quest(s) (add fire hydrant position quest #3368), by @thefeiter 

should be referencing  #3608 instead of #3368, and maybe @tiptoptom instead of me. He did the work, but I committed.

**Reply by FloEdelmann** [reply]
I fixed it in the changelog file, so it will be correct for the stable release: https://github.com/streetcomplete/StreetComplete/commit/ae6e202ceace3680b15405dd998858507f0265e1

**Reply by thefeiter** [reply]
Thanks!

**Reply by matkoniecz** [reply]
Note that you can have different commit and author identify.

`git commit --author="John Doe <john@doe.org>"`

See also https://stackoverflow.com/questions/3042437/how-to-change-the-commit-author-for-one-specific-commit

(too late for that one but may be useful in the future)

**Comment by westnordost** [comment]
I went out today myself with the smoothness quest. I must say, I am not really content with it. The more exlamation marks the more important.

### !!! Spam, difficult to answer, subjective

It adds really **a lot** of quests. Like the surface quest, only that the surface quest is in general answerable in a blink of an eye while this one requires turning it over in ones mind and still not be sure whether one made the right choice. Furthermore, for most roads, it'll be either the first or the second choice.

**Solution:** I think it should be disabled by default, mentioning in the default-disabled message that it is hard to answer (consistently) and that it is a lot. So, people who want to do it, will also be more committed to mapping this, resulting in larger "patches" of recorded data by the same contributor (see below why I think this is better)

Furthermore, this is nothing new, but it is quite subjective, still and always will be. Some people will interpret the "still ok with" very strictly like [a teacher that only gives an A if the student got 100%](https://wiki.openstreetmap.org/wiki/File_talk:Asphalt_-_good.jpg), others will score a road as still `bad` unless the potholes are so deep that they'll rip open the oil tank. Just two extreme examples.
Point is, the more people are involved in tagging it on the side and without having a larger overview, the more we'll have some kind of edit war with fluctuating `smoothness` values maybe without those participating even realizing it (because they don't see it).
I think to be most useful, `smoothness` values should be comparable _**at least**_ in one area. So if we have two identical cities and in one, someone who is strict tagged things and in the other, someone else tagged everything one category better, it's not as bad as if we had a patchwork of of `smoothness` values all tagged by people with slightly different understandings what constitutes `good` etc.

**Long term solution:** Remove this as a quest altogether and offer to tag this **exclusively** with the to be implemented layers-feature, which will then also show color-coded how the other roads have been tagged so far, when they last have been checked etc.

### !!! `good` compacted?

According to the wiki, `good` should be ok for kick scooters. I see children with kick scooters in the park on compacted ways all the time. "Good" compacted ways are definitely good for kick scooters and racing bikes, i.e. without loose gravel and even. An example way that got me thinking was
![20220112_132308](https://user-images.githubusercontent.com/4661658/149155093-0a9fd57e-be77-4cf7-97ae-59bf10d3b485.jpg)
because it is really about as smooth as the `good` paving stones in the background.

I also see those new electric scooters, though they have suspension and bigger wheels and thus are probably more comparable with city bikes (`intermediate`) in terrain compatibility or at least somewhere in between.

**Solutions:**
- Find a picture for `good` compacted surface. It should have no unevenness, no loose gravel and a relatively fine granularity of the sand/gravel mix.
- The current `intermediate` picture for compacted should probably be switched out with one that obviously **has** some loose gravel or rougher granularity.
- The wheels on the kick scooter icon should probably be a little smaller, so that it is clear that not those electrict kick scooters are meant

 ### !! _Really_ difficult to decide between `intermediate` and `bad` sett in my area

In my area, there are a lot of these wheathered chiclet-like large sett stones, similar to the ones in the picture but some of those have very shallow gaps in between and seem generally quite flat. And there are obviously still differences - some of these roads are much worse to cycle on than others. After some consideration, I came to the conclusion that they should probably still all be tagged as `bad`, because:

- `intermediate` is for **city** bikes and `bad` is for trekking bikes. The assumed difference is probably that the latter have fork suspension or even cyclocross tyres.
- the in-app description matches even for the slightly better ones
- an asphalt road would have to have really a lot of cracks and unevenness to be as bumpy as such a sett road
- the particular `bad` road I had in mind that would be far worse than those others but also be `bad` is maybe `very_bad` because there is even a warning sign "road damage"

**Solutions:**
- make it more clear that it is a city bike on `intermediate`: Maybe change the bicycle icon so that it looks more like ![holland](https://user-images.githubusercontent.com/4661658/149159084-02b31a44-8900-40ce-b8b7-5e9e7b64f6db.jpg)
- maybe find an even more obviously `very_bad` picture for `sett`, to make it clear that normal undamaged `sett` will probably not be in that category


 ### ! `good` asphalt picture too good, `intermediate` picture difficult to evaluate

I think the `good` asphalt picture is misleadingly too good. The mere existence of patches doesn't make a road less good, the patches can be smooth too. The picture makes it look like if there is a patch, it can't be `excellent` anymore.
The `intermediate` picture on the other hand shows just one spot. It is difficult to judge the big picture from that. Should the whole road look like that? That would put 99.9% of asphalt roads (I'd guess) into the `good` category.

**Solution:** Find better pictures

**Reply by mnalis** [reply]
- spam - I've noticed that too inside the city, mostly at the road crossings - both for `surface` and for `smoothness` quests - especially in areas where city has been micromapped (eg. _half a metre or footway, then kerb, then two meters of footway crossing, then kerb, then half a metre footway, than tramway crossing, then island, then half a metre footway etc... times 4 for all 4 crossings, + footway sidewalks leading to them_). In the rural areas such problems however happen very rarely (_and_ the smoothness itself is more useful there!)

  I don't know how resource expensive / complicated it is, but perhaps **very short segments** (<5m? 10m? maybe only those connecting to crossings?) **should be skipped**, both because it is spammy and of low usefulness, and it is especially confusing/subjective at such short ways (eg. _is 3 meters long footway crossing with perfect asphalt but which contain 2 high kerbs `excellent` or `very_bad`?_ Those are only "sporadic obstacles" as mentioned, but at the same time they are large percentage of length of those ways, so it is "average" too!)

  I would skip such very short ways  for `smoothness` quest definitely, but maybe also for `surface` (eg. often `asphalt` _partly_ becomes `paving_stones` or `concrete` at crossings, and it is not really useful to micromap it to that detail). Maybe even for `lit`, as it is often spammy at such locations in my experience?

- subjective - a little. But yes, I would probably agree that `smoothness` should be disabled by default. It is likely better if only persons especially interested in mapping it (like myself :smile:) enable it manually. I wouldn't remove it completely even long-term, though.

- sett - should be `bad` as reasoned, I agree

**Reply by matkoniecz** [reply]
I also opened https://github.com/streetcomplete/StreetComplete/issues/3639 - descriptions for paths are a bit confusing.

I also have some `smoothness=excellent surface=compacted` path in parks and right now it is impossible to mark them.

**Reply by westnordost** [reply]
Could you maybe make photos of those? IMO the missing option for `good` smoothness is currently a blocker for non-beta v39.0

**Reply by matkoniecz** [reply]
I am planning. I have not mentione it from start as weather is terrible in all aspects right now.

**Comment by smichel17** [comment]
> Long term solution: Remove this as a quest altogether and offer to tag this exclusively with the to be implemented layers-feature, which will then also **show color-coded how the other roads have been tagged so far**, when they last have been checked etc.

Is the part I bolded do-able within the framework of the quest, similar to how we currently show nearby elements for other quests? That could be a good medium-term solution (and hopefully it would be written in a way that the code could be re-used in the _layers_ implementation).

**Reply by westnordost** [reply]
Currently not, theoretically technically yes, but then we are already halfway at the layers implementation. Also, for `smoothness`, the **bigger** picture matters, not only the immediately surrounding ways.

**Comment by smichel17** [comment]
> * perhaps **very short segments** (<5m? 10m? maybe only those connecting to crossings?) **should be skipped**, both because it is spammy and of low usefulness, and it is especially confusing/subjective at such short ways

This is a bit meta, but it prompts an interesting question: How does this work with the idea of "differs along the way"? In theory, one might split the way to mark just the section with potholes as bad. But in practice this is probably not useful (it's only worth splitting the way **at junctions with another road**, because that affects routing possibilities).

**Reply by mnalis** [reply]
It is useful in practice, because if there is `2km` segment of the road between two crossings, which is `smoothness=good` but there is `100m` with `smoothness=horrible`, it is quite different for users than if all `2km` were horrible... So IMHO it should be mapped what is really on the ground (or not mapped at all if one deems it not important - eg. if horrible part was `2m`  one might map it as single-node obstacle instead)
